includes/actions/update-next-payment.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace AutomateWoo_Subscriptions;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Change a subscription's next payment date.
+ *
+ *
+ * @class Action_Subscription_Update_Next_Payment
+ * @since 1.2.2
+ */
+class Action_Subscription_Update_Next_Payment extends Abstract_Action_Subscription {
+
+	/**
+	 * The next payment date of the subscription before running the action.
+	 *
+	 * @var string
+	 */
+	private $original_payment_date;
+
+	/**
+	 * Explain to store admin what this action does via a unique title and description.
+	 */
+	public function load_admin_details() {
+		parent::load_admin_details();
+		$this->title       = __( 'Update Next Payment', 'automatewoo-subscriptions' );
+		$this->description = __( 'Change a subscription\'s next payment date.', 'automatewoo-subscriptions' );
+	}
+
+	/**
+	 * Edit the item managed by this class on the subscription passed in the workflow's trigger
+	 *
+	 * @throws \Exception When there is an error.
+	 */
+	public function edit_subscription( $subscription ) {
+		$this->original_payment_date = $subscription->get_date( 'next_payment' );
```

Comment:
I'd prefer if we didn't need to store old payment date as a class property. It seems like the structure in `Abstract_Action_Subscription` class could be simplified to avoid having to store as a prop. I can see that `Abstract_Action_Subscription` is based on the pre-existing `\AutomateWoo\Action_Subscription_Edit_Item_Abstract` class but I'm happy to use a different structure here. I think moving the `run()` method from the parent into the child class could help simplify things.

Reply:
> I'd prefer if we didn't need to store old payment date as a class property.

I've just removed the class property –– we were using it to specify the old payment date on the subscription notes (not doing that anymore).


>  I can see that `Abstract_Action_Subscription` is based on the pre-existing `\AutomateWoo\Action_Subscription_Edit_Item_Abstract` class

Ahh, I got trapped by the name of the class (`Action_Subscription_Edit_Item_Abstract`) –– I thought it's used to update the subscription items (line/coupon/shipping).


> I think moving the run() method from the parent into the child class could help simplify things.

Not going that route for now.

<hr>

On a side note, the only thing that I'm not liking about this is the timezone and the format of the new payment date shown on the subscription note. The timezone is UTC and the format is `mysql`. Honestly, it confused me when I was looking at the subscription notes on a real WooCommerce store on which we ran this. Let me know what you think about this.

![image](https://user-images.githubusercontent.com/11808845/102893773-20b66f00-4488-11eb-80f9-0e6b2eb39143.png)

Reply:
> On a side note, the only thing that I'm not liking about this is the timezone and the format of the new payment date shown on the subscription note

I agree, it would be better displayed in the site's timezone and formatted with `wc_date_format()` and `wc_time_format()`.